fv3core/stencils/dyn_core.py

```diff
@@ -179,381 +197,415 @@ def dyncore_temporaries(shape):
-            if spec.namelist.use_old_omega:  # apparently True
-                set_pem(
-                    state.delp,
-                    state.pem,
-                    state.ptop,
-                    origin=(grid.is_ - 1, grid.js - 1, 0),
-                    domain=(grid.nic + 2, grid.njc + 2, grid.npz),
-                )
-        if global_config.get_do_halo_exchange():
-            reqs_vector.wait()
-            if not hydrostatic:
-                reqs["w_quantity"].wait()
-
-        # compute the c-grid winds at t + 1/2 timestep
-        state.delpc, state.ptc = c_sw.compute(
-            state.delp,
-            state.pt,
-            state.u,
-            state.v,
-            state.w,
-            state.uc,
-            state.vc,
-            state.ua,
-            state.va,
-            state.ut,
-            state.vt,
-            state.divgd,
-            state.omga,
-            dt2,
+        self._set_pem = FixedOriginStencil(
+            set_pem,
+            origin=self.grid.compute_origin(add=(-1, -1, 0)),
+            domain=self.grid.domain_shape_compute(add=(2, 2, 0)),
         )
 
-        if spec.namelist.nord > 0 and global_config.get_do_halo_exchange():
-            reqs["divgd_quantity"] = comm.start_halo_update(
-                state.divgd_quantity, n_points=utils.halo
-            )
-        if not hydrostatic:
-            if it == 0:
-                if global_config.get_do_halo_exchange():
-                    reqs["gz_quantity"].wait()
-                copy_stencil(
-                    state.gz,
-                    state.zh,
-                    origin=grid.full_origin(),
-                    domain=grid.domain_shape_full(add=(0, 0, 1)),
-                )
-            else:
-                copy_stencil(
-                    state.zh,
-                    state.gz,
-                    origin=grid.full_origin(),
-                    domain=grid.domain_shape_full(add=(0, 0, 1)),
+        pgradc_origin = self.grid.compute_origin()
+        pgradc_domain = self.grid.domain_shape_compute(add=(1, 1, 0))
+        ax_offsets = axis_offsets(self.grid, pgradc_origin, pgradc_domain)
+        pgradc_kwargs = {
+            "origin": pgradc_origin,
+            "domain": pgradc_domain,
+            "externals": {"hydrostatic": self.namelist.hydrostatic, **ax_offsets},
+        }
+        self._p_grad_c = FixedOriginStencil(p_grad_c_stencil, **pgradc_kwargs)
+        self._initialize_data = FixedOriginStencil(
+            initialize_data,
+            origin=self.grid.full_origin(),
+            domain=self.grid.domain_shape_full(),
+            externals={"hydrostatic": self.namelist.hydrostatic},
+        )
+
+    def __call__(self, state):
+        # u, v, w, delz, delp, pt, pe, pk, phis, wsd, omga, ua, va, uc, vc, mfxd,
+        # mfyd, cxd, cyd, pkz, peln, q_con, ak, bk, diss_estd, cappa, mdt, n_split,
+        # akap, ptop, pfull, n_map, comm):
+        grid = self.grid
```

Comment:
Can you delete this line and use `self.grid` below instead? That would avoid a reader needing to question whether this came from an argument, global state, or self.